uaclient/cli.py

```diff
@@ -1072,36 +1038,17 @@ def _get_contract_token_from_cloud_identity(cfg: config.UAConfig) -> str:
             raise exceptions.UserFacingError(
                 ua_status.MESSAGE_DETACH_AUTOMATION_FAILURE
             )
-    contract_client = contract.UAContractClient(cfg)
-    try:
-        tokenResponse = contract_client.request_auto_attach_contract_token(
-            instance=instance
-        )
-    except contract.ContractAPIError as e:
-        if e.code and 400 <= e.code < 500:
-            raise exceptions.NonAutoAttachImageError(
-                ua_status.MESSAGE_UNSUPPORTED_AUTO_ATTACH
-            )
-        raise e
-    if current_iid:
-        cfg.write_cache("instance-id", current_iid)
-
-    return tokenResponse["contractToken"]
 
-
-@assert_root
-@assert_lock_file("ua auto-attach")
-def action_auto_attach(args, *, cfg):
-    disable_auto_attach = util.is_config_value_true(
-        config=cfg.cfg, path_to_value="features.disable_auto_attach"
-    )
-    if disable_auto_attach:
-        msg = "Skipping auto-attach. Config disable_auto_attach is set."
-        logging.debug(msg)
-        print(msg)
+    try:
+        actions.auto_attach(cfg, instance)
+    except util.UrlError:
+        print(ua_status.MESSAGE_ATTACH_FAILURE)
+        return 1
+    except exceptions.UserFacingError:
```

Comment:
Sorry @orndorffgrant, I think I am still missing something. Shouldn't we still surface some error message to the user in that situation ?

Reply:
@lucasmoura This is just doing the same thing that it was doing before but rearranged. The logging now happens in actions.attach_with_token
It used to look like this in `cli._attach_with_token`
```python
    try:
        contract.request_updated_contract(
            cfg, token, allow_enable=allow_enable
        )
    except util.UrlError as exc:
        with util.disable_log_to_console():
            logging.exception(exc)
        print(ua_status.MESSAGE_ATTACH_FAILURE)
        cfg.status()  # Persist updated status in the event of partial attach
        config.update_ua_messages(cfg)
        return 1
    except exceptions.UserFacingError as exc:
        # *********************** logged here *************************************
        logging.warning(exc.msg)
        cfg.status()  # Persist updated status in the event of partial attach
        config.update_ua_messages(cfg)
        return 1
```

Now it looks like this.

In `actions.attach_with_token`:
```python
    try:
        contract.request_updated_contract(
            cfg, token, allow_enable=allow_enable
        )
    except util.UrlError as exc:
        with util.disable_log_to_console():
            LOG.exception(exc)
        # *********************** notice no print statement here *************************************
        cfg.status()  # Persist updated status in the event of partial attach
        config.update_ua_messages(cfg)
        # *********************** raise instead of return *************************************
        raise exc
    except exceptions.UserFacingError as exc:
        # *********************** I left this log here *************************************
        LOG.warning(exc.msg)
        cfg.status()  # Persist updated status in the event of partial attach
        config.update_ua_messages(cfg)
        # *********************** raise instead of return *************************************
        raise exc
```

In `cli.action_attach`:
```python
    try:
        actions.attach_with_token(
            cfg, token=args.token, allow_enable=args.auto_enable
        )
    except util.UrlError:
        # *********************** the print happens at this level now *************************************
        print(ua_status.MESSAGE_ATTACH_FAILURE)
        # *********************** same with return for both cases *************************************
        return 1
    except exceptions.UserFacingError:
        # *********************** same with return for both cases *************************************
        return 1
```

Arguably that UserFacingError should not be logged in `actions.attach_with_token`. I was just trying to keep everything exactly the same for this refactor. But I can add another commit to this PR to let the UserFacingError bubble up instead of handling like it is now.

Reply:
Thanks so much for the whole context here. My mistake that I missed that we are actually logging the error.
I think we can keep it as is